Modules/Common/AptosLikeMainModule.php

```diff
@@ -0,0 +1,297 @@
+                        if ($balance === '0')
+                        {
+                            continue;
+                        }
+
+                        $events[] = [
+                            'block' => $block['block_height'],
+                            'transaction' => $trx['hash'],
+                            'time' => $this->block_time,
+                            'address' => $address,
+                            'sort_key' => $sort_key++,
+                            'effect' => $balance,
+                            'failed' => $failed,
+                            'extra' => null,
+                        ];
+
+                        $events[] = [
+                            'block' => $block['block_height'],
+                            'transaction' => $trx['hash'],
+                            'time' => $this->block_time,
+                            'address' => 'the-void',
+                            'sort_key' => $sort_key++,
+                            'effect' => '-' . $balance,
+                            'failed' => $failed,
+                            'extra' => null,
+                        ];
+                    }
+
+                    break;
+
+                case 'block_metadata_transaction':
+                    // Process the validators rewards for past epoch.
+                    // Currently, slashing is not implemented: https://aptos.dev/concepts/staking/
+                    foreach ($trx['events'] as $trx_event)
+                    {
+                        if ($trx_event['type'] === '0x1::stake::DistributeRewardsEvent')
+                        {
+                            $events[] = [
+                                'block' => $block['block_height'],
+                                'transaction' => $trx['hash'],
+                                'time' => $this->block_time,
+                                'address' => $trx_event['data']['pool_address'],
+                                'sort_key' => $sort_key++,
+                                'effect' => $trx_event['data']['rewards_amount'],
+                                'failed' => $failed,
+                                'extra' => AptosSpecialTransactions::ValidatorReward->value,
+                            ];
+
+                            $events[] = [
+                                'block' => $block['block_height'],
+                                'transaction' => $trx['hash'],
+                                'time' => $this->block_time,
+                                'address' => 'the-void',
+                                'sort_key' => $sort_key++,
+                                'effect' => '-' . $trx_event['data']['rewards_amount'],
+                                'failed' => $failed,
+                                'extra' => AptosSpecialTransactions::ValidatorReward->value,
+                            ];
+                        }
```

Comment:
else ..?

Reply:
Else there is no validators rewards in this block. `block_metadata_transaction` type contains only validator rewards events at every epoch changed.